### DISCUSSION TITLE
Command Operations for the Linux Version of APSIM – User Guide

### DISCUSSION THREAD

**Discussion by lrc756866304** [APSIMInitiative/ApsimX#9392] [Q&A]
![图片444](https://github.com/user-attachments/assets/2807f200-0c33-4252-9c8a-ae03fb115f95)
![图片333](https://github.com/user-attachments/assets/3851085b-6f6c-4a17-bb03-d9e8734d1e6c)
![图片222](https://github.com/user-attachments/assets/91040c9d-4b2c-47b9-ac18-3d21e9e7844a)
![图片111](https://github.com/user-attachments/assets/d389ce5e-a35d-4965-bad8-0d3346943d51)

I want to use the Linux version of APSIM for simulations. I have successfully configured the APSIM image on the HPC system and can run an .apsimx file to generate the corresponding results (.db file). However, I couldn't find documentation or examples related to the command operations, and many commands I tried have failed. Below are some of the results from my attempts:

I tried modifying an .apsimx file following the instructions from a .txt file. However, I am not sure about the correct format of the information that should be in the .txt file. There was no explanation provided regarding that.

The --upgrade command is supposed to upgrade a file to the latest version of the .apsimx file format without running the simulation. Does this upgrade the .apsimx file itself to a newer version? I thought that .apsimx files do not have versions, and only the APSIM software versions differ.

The --run-tests command mentions running all tests inside the file after executing the simulation. I do not quite understand what this means. Are there different tests embedded within an .apsimx file?

The --verbose option is supposed to print detailed messages to the terminal when a simulation starts or finishes. However, after running it, I still only obtained a .db result file without any additional detailed messages.

The --csv option is supposed to export all reports into .csv files. However, after running the command, it still only generated a .db file, and I did not find any .csv files.

Regarding the --merge-db-files command, I could not find any examples or explanations for this operation, so I am unsure about how to use it and what format to expect.

The --list-simulations option should list the simulation names without running them. However, after running the command, it still only generated a .db result file instead of listing the simulation names.

In short, I am unable to use these commands flexibly, and I can't find any relevant tutorials or examples. I hope to get some help; perhaps there are related operation documents or manuals? There are very few tutorials available for using APSIM on Linux systems, and this has troubled me for quite some time. I am truly grateful to anyone who takes the time to look into this issue.

**Comment by ric394** [comment]
Hi @lrc756866304,
I believe I can help you here.

Documentation for command line usage of ApsimX can be found here: [command line docs](https://apsimnextgeneration.netlify.app/usage/commandline/)

1. For your first issue about the format of your config file you can find an explanation here:  [Example config file](https://apsimnextgeneration.netlify.app/usage/commandline/command-line-how-to/#what-does-an-example-config-file-look-like)
2. Apsimx files have a version that defines the structure of the data in the file. This changes as parameters of models are changed over time. It does not directly relate to the Apsim program version.
3. Regarding `-run-tests` switch more information can be found here: [run-tests switch docs](https://apsimnextgeneration.netlify.app/usage/tests/)
4. The `--verbose` switch will output specific messages as files are run. If you did not see any detailed messages, it's likely there was an issue running the file.
5. Similar to above, this will only output csv files if data was generated and available. Make sure the file is running to completion.
6. The `--list-simulations` switch works similar to the below image, this file just had a single Simulation:

![image](https://github.com/user-attachments/assets/da2c3106-ca9b-41b0-8848-6dfec353a2e2)

I hope these explanations are helpful. Also if you have any feedback about the docs or how to improve them that would be very helpful.

**Reply by lrc756866304** [reply]
Thank you very much for your response. Furthermore, I would like to provide the commands I used, as perhaps there might be an error in the format? Below are the commands that failed:

./apsim run --run-tests /work/home/Examples/Wheat.apsimx
./apsim run --verbose /work/home/Examples/Wheat.apsimx
./apsim run --csv /work/home/Examples/Wheat.apsimx
./apsim run --list-simulations /work/home/Examples/Wheat.apsimx

All of these commands only generated a .db format result. I did not get any additional output, and the result was the same as when I ran 

./apsim run /work/home/Examples/Wheat.apsimx.

**Reply by ric394** [reply]
Instead of `apsim` use `models`.
`models` has all the additional switches/flags that you're using in your example and is the usual way to run APSIM via command line. `apsim` only has a very restricted set of switches/flags.

**Reply by lrc756866304** [reply]
Thank you very much for your help. The command-based approach for Models works well. As a beginner, I find it difficult to fully understand and implement a lot of the official information, and I really appreciate your warm assistance. I feel that there may still be too few examples in the Linux tutorials to help beginners better understand. Perhaps for those more technically proficient, the current tutorials are sufficient, but more examples would benefit beginners like me.

**Reply by ric394** [reply]
Could I ask what further instructions would be helpful to your use cases? I may be able to create suitable tutorials/guides.
